source/common/ssl/ssl_socket.cc

```diff
@@ -339,6 +339,30 @@ std::string SslSocket::protocol() const {
   return std::string(reinterpret_cast<const char*>(proto), proto_len);
 }
 
+std::string SslSocket::getSerialNumberFromCertificate(X509* cert) const {
```

Comment:
A few style items:

* Make this a static member, or not a member of the class. It doesn't use any class data. 
* Pass an X509& instead of X509* to indicate that NULL isn't valid.

As you noted in your comment, it would be nice to deduplicate this code. Can you create an ssl utility namespace for stuff like this and put this function in it? I don't think you need to move all the duplicate code there in this PR; let's just fix what's being changed already in this PR.

Reply:
Agreed on both style items. The original reason I made this an instance member which is passed a pointer was for consistency with other member functions, such as `getUriSanFromCertificate()` and `getSubjectFromCertificate()`.
Now that the function is being moved the `Ssl::Utility` namespace, it can simply be made global.

Reply:
@ggreenway, see commit 77235c1, in which I created an `Ssl::Utility` namespace and moved this function into it.